eos-grub-upgrade

```diff
@@ -0,0 +1,276 @@
+# | B    | Standalone, since 3.0.0 and before 3.1.0 | EFI-64 / GPT  | GRUB_AMD64: \EFI\BOOT\grubx64.efi in ESP on same disk as /    |
+# |      |                                          |               | GRUB_IA32:  none                                              |
+# |      |                                          |               | SHIM:       \EFI\BOOT\bootx64.efi                             |
+# +------+------------------------------------------+---------------+---------------------------------------------------------------+
+# | C    | Standalone, before 3.0.0                 | EFI-64 / GPT  | GRUB:       \EFI\BOOT\bootx64.efi in ESP on same disk as /    |
+# |      |                                          |               | GRUB_IA32:  none                                              |
+# |      |                                          |               | SHIM:       none                                              |
+# +------+------------------------------------------+---------------+---------------------------------------------------------------+
+
+ESP_MOUNTPOINT=/boot/efi
+
+##########################################################################
+## UEFI
+##########################################################################
+
+## AMD64 GRUB
+declare -A GRUB_AMD64=(
+  [A]="${ESP_MOUNTPOINT}/EFI/endless/grubx64.efi"
+  [B]="${ESP_MOUNTPOINT}/EFI/BOOT/grubx64.efi"
+  [C]="${ESP_MOUNTPOINT}/EFI/BOOT/bootx64.efi"
+)
+NEW_GRUB_AMD64=/usr/lib/grub/images/x86_64-efi/grubx64.efi
+
+## SHIM
+declare -A SHIM=(
+  [A]="${ESP_MOUNTPOINT}/EFI/BOOT/bootx64.efi"
+  [B]="${ESP_MOUNTPOINT}/EFI/BOOT/bootx64.efi"
+)
+NEW_SHIM=/usr/lib/shim/shim.efi
+
+SHIM_COPY=${ESP_MOUNTPOINT}/EFI/endless/shim.efi
+NEW_SHIM_COPY=/usr/lib/shim/shim.efi
+
+## IA32 GRUB
+GRUB_IA32=${ESP_MOUNTPOINT}/EFI/BOOT/bootia32.efi
+NEW_GRUB_IA32=/usr/lib/grub/images/i386-efi/bootia32.efi
+
+## MokManager
+MOKMANAGER=${ESP_MOUNTPOINT}/EFI/endless/MokManager.efi
+NEW_MOKMANAGER=/usr/lib/shim/MokManager.efi
+
+## fallback
+FALLBACK=${ESP_MOUNTPOINT}/EFI/BOOT/fallback.efi
+NEW_FALLBACK=/usr/lib/shim/fallback.efi
+
+## BOOT.CSV
+BOOTCSV=${ESP_MOUNTPOINT}/EFI/endless/BOOT.CSV
+NEW_BOOTCSV=/usr/lib/shim/BOOT.CSV
+
+## grub.cfg
+GRUB_CFG=/boot/grub/grub.cfg
+NEW_GRUB_CFG=/usr/lib/grub/conf/grub.cfg
+
+EFI_LIST=( GRUB_AMD64 SHIM SHIM_COPY GRUB_IA32 MOKMANAGER FALLBACK BOOTCSV GRUB_CFG )
+
+##########################################################################
+
+umount_esp() {
+  umount ${ESP_MOUNTPOINT} 2> /dev/null
+}
+trap umount_esp EXIT
+
+get_last_char() {
+  local ret=$1
+
+  while [[ ${#ret} != "1" ]]; do
+    ret=${ret#?}
+  done
+
+  echo "${ret}"
+}
+
+get_root_disk() {
+  local root_part
+  root_part=$(systemctl show -p What sysroot.mount)
+  root_part=${root_part#What=}
+
+  if [[ -z ${root_part} ]]; then
+    echo "Couldn't identify root device"
+    exit 1
+  fi
+
+  # Get the root disk device file
+  local root_disk=
+  case ${root_part} in
+    /dev/sd??)
```

Comment:
Could do this check the other way around: if the path doesn't match `/dev/*p[0-9]` then just chop off the last character (and don't explicitly match `/dev/sd??`). I don't know the full menagerie of device name formats though.